### PR TITLE
Add graftorio

### PR DESCRIPTION
**Pull request by oof2win2**
Partially add some bits of [graftorio](https://github.com/TheVirtualCrew/graftorio). Data can be collected with the `/collectdata` command and is only sent to RCON currently.
See [this](https://github.com/oof2win2/AwF-Bot/blob/master/src/base/GrafanaHandler.ts) for implementation of handling the data in JS.

### CODE REVIEW

File: modules/graftorio/require.lua [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+Commands.new_command("collectdata", "Collect data for RCON usage")
+		:add_param("location", true)
+		:register(function()
+			-- this must be first as it overwrites the stats
+			-- also makes the .other table for all forces
+			statics.collect_statics()
+			general.collect_other()
+			if config.modules.forcestats then
+				forcestats.collect_production()
+			end
```
Comment (Cooldude2606) [inline modules/graftorio/require.lua:20]: `collect_loginet` is never called which is part of the `forcestats` module.
`general` has a config which is never tested for.
```suggestion
			statics.collect_statics()
			if config.modules.general then
			  general.collect_other()
			end
			if config.modules.forcestats then
				forcestats.collect_production()
				forcestats.collect_loginet()
			end
```

File: modules/graftorio/require.lua [hunk 2]
```diff
@@ -0,0 +1,23 @@
+local Commands = require("expcore.commands")
+local config = require("config.graftorio")
+
+local general = require("modules.graftorio.general")
```
Comment (Cooldude2606) [inline modules/graftorio/require.lua:4]: `general` needs a similar test which forcestats has.
```suggestion
local general; if config.modules.general then
  general = require("modules.graftorio.general")
end
```

**Review by Cooldude2606 — CHANGES_REQUESTED** [review]
Two small corrections required. I will also have mark review this PR due to his familiarity with grafana.

**Review by Cooldude2606 — APPROVED** [review]
Changes have been addressed. LGTM.

File: config/expcore/roles.lua [hunk 3]
```diff
@@ -52,6 +52,7 @@ Roles.new_role('Administrator','Admin')
     'gui/warp-list/bypass-cooldown',
     'gui/warp-list/bypass-proximity',
     'command/connect-all',
+		'command/collectdata'
```
Comment (mark9064) [inline config/expcore/roles.lua:55]: Might just be github but there seems to be a whitespace issue here?

File: config/graftorio.lua [hunk 4]
```diff
@@ -0,0 +1,7 @@
+return {
+	modules = {
+		["forcestats"] = true,
+		["logistorage"] = false,
```
Comment (mark9064) [inline config/graftorio.lua:4]: We aren't shipping the `logistorage` module, so we can drop this for now?

**Review by mark9064 — COMMENTED** [review]
Merged just as I started the review, oh well! LGTM overall anyway.
One potentially important thing: we probably need to acknowledge the graftorio license if we are using their code here directly.

File: config/graftorio.lua (hunk 4 above)
Comment (oof2win2) [reply config/graftorio.lua:4]: logistorage itself is a part of forcestats collected with robot metrics. might be worth splitting it apart in the future

File: config/expcore/roles.lua (hunk 3 above)
Comment (oof2win2) [reply config/expcore/roles.lua:55]: you are right there.

**Comment by oof2win2** [comment]
We don't need to acknowledge the graftorio licence as this is basically a rewrite of it that is suitable for the scenario and doesn't have features that I deemed unnecessary, such as translation of prototype names. Graftorio itself by TheVirtualCrew also outputs into a file. It would have been more appropriate to add the licence if #222 was merged since that was mainly their code, but it wasn't merged and this is a rewrite.

**Pull request merged**